core/modules/user/user.module

```diff
@@ -1630,6 +1636,18 @@ function user_menu_alter(&$items) {
   }
 }
 
+/**
+ * Return the login page appearance setting.
+ */
+function user_login_appearance_setting() {
+  $appearance_setting = config_get('system.core', 'user_login_appearance_setting');
+  // If Layout module is disabled, use links.
+  if ($appearance_setting == 'simplified' && !function_exists('layout_suppress')) {
```

Comment:
It still feels odd to me, to create a whole function only because one (unrelated) functional test that disables the layout module (??) fails with a missing function otherwise.

If we could solve that with another approach, we could simply use `config_get('system.core', 'user_login_appearance_setting');` - and system.core is statically cached.

Reply:
I agree this is strange. As @docwilmot pointed out, there is a test (`testMainContentFallback`) that disables layout - a module that can't be disabled through the UI. 

What do you suggest? I guess we could modify `testMainContentFallback` to use the "tabs" look for running the test, so that `user_init()` doesn't try to use `layout_suppress()`.

Reply:
On second thoughts, I'm not sure if modifying the test may be a good idea. Are there modules or themes out there that disable `layout`? If that's the case, then checking for the existence of function `layout_suppress` is wiser.

Reply:
> On second thoughts, I'm not sure if modifying the test may be a good idea.

I took a look - this test has been touched last time when the Layout module has been added to Backdrop. So the test actually predates the module (2009).
Personally, I'd have no concerns adapting this test. But maybe I'm missing something substantial.

Reply:
I'm inclined to also modify the test - after all this test doesn't actually test the situation when the layout module is disabled. It disables the layout module to preserve the original intent of the test, which predates that module. I'll give people a bit more time to comment here before making the modifications to the test.

Reply:
OK, I opted for a new approach that avoids `hook_init`, and therefore this hack to accommodate the weird test is not needed anymore. Resolving.